extension/content/tests/components/recipes/details/RecipeDetails.test.js

```diff
@@ -235,4 +234,92 @@ describe("The `RecipeDetailForm` component", () => {
 
     expect(NormandyAPI.prototype.closeApprovalRequest).toBeCalled();
   });
+
+  it("should be able to pause recipes", async () => {
+    const nextRevisionId = 10042;
+    jest
+      .spyOn(NormandyAPI.prototype, "patchRecipe")
+      .mockImplementation(async () => ({
+        latest_revision: { id: nextRevisionId },
+      }));
+    jest
+      .spyOn(NormandyAPI.prototype, "requestApproval")
+      .mockImplementation(async () => approvalRequestFactory.build());
+
+    const recipe = recipeFactory.build({
+      latest_revision: {
+        action: { name: "multi-preference-experiment" },
+        arguments: { isEnrollmentPaused: false },
+      },
+    });
+    setup(recipe);
+
+    const doc = await render(<App />);
+
+    // Navigate to the detail page
+    fireEvent.click(doc.getByText("Recipes"));
+    fireEvent.click(await doc.findByText(recipe.latest_revision.name));
+
+    // wait for load to complete
+    await doc.findByText(recipe.latest_revision.name);
+
+    const pauseButton = await doc.findByText("Pause");
+    expect(pauseButton).toBeInTheDocument();
+
+    fireEvent.click(pauseButton);
+    expect(NormandyAPI.prototype.patchRecipe).toBeCalledWith(recipe.id, {
+      comment: expect.any(String),
+      arguments: {
+        ...recipe.latest_revision.arguments,
+        isEnrollmentPaused: true,
+      },
+    });
+    // Give it a moment for the event loop to spin.
```

Comment:
Shouldn't we assert that button is disabled during request? Does that imply introducing delya in `mockImplementation(...)`?